File: Medium/2026-06-07-2196-Create-Binary-Tree-From-Descriptions/solution.js
class TreeNode {
    constructor(val) {
        this.val = val;
        this.left = null;
        this.right = null;
    }
}

var createBinaryTree = function(descriptions) {
    const map = new Map();
    const hasParent = new Set();

    for (const [parent, child, isLeft] of descriptions) {
        if (!map.has(parent)) {
            map.set(parent, new TreeNode(parent));
        }
        if (!map.has(child)) {
            map.set(child, new TreeNode(child));
        }
        
        if (isLeft === 1) {
            map.get(parent).left = map.get(child);
        } else {
            map.get(parent).right = map.get(child);
        }
        
        hasParent.add(child);
    }

    for (const [parent] of descriptions) {
        if (!hasParent.has(parent)) {
            return map.get(parent);
        }
    }
};

File: Medium/2026-06-07-2196-Create-Binary-Tree-From-Descriptions/solution.java
import java.util.HashMap;
import java.util.Map;

class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode(int x) { val = x; }
}

public class Solution {
    public TreeNode createBinaryTree(int[][] descriptions) {
        Map<Integer, TreeNode> nodeMap = new HashMap<>();
        Set<Integer> childNodes = new HashSet<>();
        
        for (int[] desc : descriptions) {
            int parentVal = desc[0];
            int childVal = desc[1];
            int isLeft = desc[2];
            
            TreeNode parentNode = nodeMap.getOrDefault(parentVal, new TreeNode(parentVal));
            TreeNode childNode = nodeMap.getOrDefault(childVal, new TreeNode(childVal));
            
            if (isLeft == 1) {
                parentNode.left = childNode;
            } else {
                parentNode.right = childNode;
            }
            
            nodeMap.put(parentVal, parentNode);
            nodeMap.put(childVal, childNode);
            childNodes.add(childVal);
        }
        
        for (int key : nodeMap.keySet()) {
            if (!childNodes.contains(key)) {
                return nodeMap.get(key);
            }
        }
        
        return null; // This line should never be reached if the input is valid
    }
}